src/Features/CSharp/Portable/ConvertLinq/CSharpConvertLinqQueryToForEachProvider.cs

```diff
@@ -0,0 +1,645 @@
+                    {
+                        if (!ValidateTypeSymbol(typeArgument))
+                        {
+                            return false;
+                        }
+                    }
+                }
+
+                return true;
+            }
+
+            private static bool TryGetTypeSyntax(ITypeSymbol typeSymbol, out TypeSyntax typeSyntax)
+            {
+                if (!ValidateTypeSymbol(typeSymbol))
+                {
+                    typeSyntax = default;
+                    return false;
+                }
+
+                typeSyntax = typeSymbol.GenerateTypeSyntax();
+                return true;
+            }
+
+            private static bool TryFindParentAssignment(SyntaxNode node, out SyntaxNode result)
+            {
+                while (node != null)
+                {
+                    if (node is StatementSyntax)
+                    {
+                        result = node;
+                        return true;
+                    }
+
+                    node = node.Parent;
+                }
+
+                result = default;
+                return false;
+            }
+
+            private string GetFreeSymbolName(string prefix, int position)
+            {
+                var symbols = _semanticModel.LookupSymbols(position);
+                if (!symbols.Any(s => s.Name == prefix))
+                {
+                    return prefix;
+                }
+
+                int counter = 1;
+                do
+                {
+                    string candidate = $"{prefix}{counter}";
+                    if (!symbols.Any(s => s.Name == candidate))
+                    {
+                        return candidate;
+                    }
+                    counter++;
+                } while (true);
+            }
+
+            private bool TryConvertInForEach(QueryExpressionSyntax source, out DocumentUpdate documentUpdate)
+            {
+                ForEachStatementSyntax commonForEachStatement = (ForEachStatementSyntax)source.Parent;
+                StatementSyntax statement = commonForEachStatement.Statement;
+
+                StatementSyntax internalNodeMethod(ExpressionSyntax expression)
+                {
+                    if (expression.Kind() == SyntaxKind.IdentifierName && ((IdentifierNameSyntax)expression).Identifier.ValueText == commonForEachStatement.Identifier.ValueText)
+                    {
+                        return statement;
+                    }
+                    else
+                    {
+                        var declaration = SyntaxFactory.LocalDeclarationStatement(
+                            SyntaxFactory.VariableDeclaration(
+                                VarNameIdentifier,
+                                SyntaxFactory.SingletonSeparatedList(
+                                    SyntaxFactory.VariableDeclarator(commonForEachStatement.Identifier, null, SyntaxFactory.EqualsValueClause(expression)))));
```

Comment:
1. consider using SyntaxGenerator.  
2. use named parameters for 'null'.
3. You're hardcoding 'var', but that may not be the user tyep-style preference.

Reply:
1 & 3 -https://github.com/dotnet/roslyn/issues/25639
2 - done
thanks!

---
In reply to: [175267869](https://github.com/dotnet/roslyn/pull/25362#discussion_r175267869) [](ancestors = 175267869)